hi-ml-cpath/src/health_cpath/preprocessing/tiff_conversion.py

```diff
@@ -0,0 +1,241 @@
+from enum import Enum
+import logging
+import math
+import numpy as np
+
+from health_cpath.preprocessing.loading import WSIBackend
+from health_cpath.utils.naming import SlideKey
+from monai.data.wsi_reader import WSIReader
+from monai.transforms import MapTransform
+from openslide import OpenSlide
+from pathlib import Path
+from tifffile.tifffile import TiffWriter, PHOTOMETRIC, COMPRESSION
+from typing import Any, Dict, List, Optional, Tuple
+
+
+AMPERSAND = "&"
+UNDERSCORE = "_"
+
+
+class WSIFormat(str, Enum):
+    """The format of the wsi file."""
+    NDPI = "ndpi"
+    SVS = "svs"
+    TIF = "tif"
+    TIFF = "tiff"
+
+
+class ConvertWSIToTiffd(MapTransform):
+    """Converts a wsi file to a tiff file. The tiff file is saved in the output_folder with the same name as the src
```

Comment:
could you please add supported formats in the docstring

Reply:
Sure, this should work with all supported format by openslide